File: 7-ReverseInteger/7-ReverseInteger.java
// Last updated: 7/27/2026, 3:02:52 PM
class Solution {
    public int reverse(int x) {
        int res=0;
        
        
        while(x!=0){
            int dig=x%10;
            x=x/10;
            if(res>Integer.MAX_VALUE/10 || (res== Integer.MAX_VALUE/10 && dig>7)){
                return 0;
            }
            if(res<Integer.MIN_VALUE/10 || (res== Integer.MIN_VALUE/10 && dig<-8)){
                return 0;
            }
            res=res*10+dig;
            
            
        }   
       
            return res;
        
                             
    }
}

File: 7-ReverseInteger/7-ReverseInteger.cpp
// Last updated: 7/27/2026, 3:02:57 PM
class Solution {
public:
    int reverse(int x) {
        int revint=0;
        while(x!=0){
            int dig=x%10;
            if(revint>(INT_MAX/10) || revint<(INT_MIN/10)){
             return 0;
            }
            else{
                revint=(revint*10)+dig;
                x=x/10;
            }
            
        }
       
        return revint;
    }
};